zcash_client_backend/Cargo.toml

```diff
@@ -78,12 +78,16 @@ group.workspace = true
 orchard = { workspace = true, optional = true }
 sapling.workspace = true
 
+# - Sync engine
+tokio = { version = "1.21.0", optional = true, features = ["fs", "macros"] }
```

Comment:
I do not know where these two features are depended on. The only feature I _do_ see used is `rt`, which is not enabled here; therefore I believe this PR is currently broken for downstream users (it won't appear broken when testing locally, because `rt-multi-thread` is enabled in dev-dependencies).

Reply:
I agree "rt" makes more sense in the context of this PR and will make this change. The reason why this is here is because I am developping this PR alongside #1217 which builds upon your 1169-sync engine PR. In your PR you use these features along with a JoinHandle and i wanted to keep in line with your decisions as much as possible. You are right... now you bring this up I wonder how this is building as JoinHandle is not part of "fs" or "macros"

Reply:
To clarify, after my other comment is addressed (i.e. not using `JoinHandle`), we _shouldn't_ enable `rt` here (i.e. addressing the other comment will also fix the breakage). I also think we can remove `fs` and `macros` (at least from this PR; they can be added later if there is a later PR that needs them).